packages/@ember/-internals/glimmer/tests/integration/components/life-cycle-test.js

```diff
@@ -1192,6 +1192,11 @@ class LifeCycleHooksTest extends RenderingTestCase {
         ['nested-item', 'willRender'],
         ['nested-item', 'willInsertElement'],
 
+        ['nested-item', 'didInsertElement'],
+        ['nested-item', 'didRender'],
+        ['no-items', 'didInsertElement'],
+        ['no-items', 'didRender'],
```

Comment:
Discussed this offline, the change was caused by removing the custom curly-component destruction queue that existed on the environment before. This means that all components now call their _`didDestroy`_ hooks after their `didCreate` and `didUpdate` hooks. Before, there was divergent behavior between curly components and other types of components here.

I believe the conclusion was that this change is relatively low risk, since the timing of `didDestroy` style hooks has always been async/later and generally isn't relied as much as `willDestroy` style hooks. If we encounter issues later due to this change, we can either update Glimmer-VM directly to move the position of the queue, or we can add back the custom queue specifically for curly components.

Reply:
Thank you for summarizing our conversation on this! We will just need to monitor for any issues reported (worse case scenario we bring back a custom queue that we flush like before, but we'd much prefer to avoid that).